numba/cuda/cudadecl.py

```diff
@@ -346,6 +347,19 @@ class Cuda_fp16_unary(ConcreteTemplate):
     return Cuda_fp16_unary
 
 
+def _genfp16_unary_operator(l_key):
+    @register
+    class Cuda_fp16_unary(AbstractTemplate):
+        key = l_key
+
+        def generic(self, args, kws):
+            assert not kws
```

Comment:
Got a small question: `assert` can be manually skipped in production code with `python -O`. Is this something we want to explicitly check for? If so I would suggest check explicitly with `if ...: raise`.

Reply:
I think this comes from a pattern that already exists a couple of times in `cudadecl`, and it's not something that really adds much value - I've never seen it get hit, and I think it probably just exists because it was perhaps once useful a long time ago (maybe in NumbaPro). I think instead of asserting, we could maybe just `return None` (as in, fail to type) if `kws` is something that's not false-ish.

Reply:
I think this comes from a pattern that already exists a couple of times in `cudadecl`, and it's not something that really adds much value - I've never seen it get hit, and I think it probably just exists because it was perhaps once useful a long time ago (maybe in NumbaPro). I think instead of asserting, we could maybe just `return None` (as in, fail to type) if `kws` is something that's not false-ish.